vllm/entrypoints/openai/protocol.py

```diff
@@ -601,8 +744,10 @@ class ChatCompletionResponseChoice(OpenAIBaseModel):
     index: int
     message: ChatMessage
     logprobs: Optional[ChatCompletionLogProbs] = None
-    finish_reason: Optional[str] = None
-    stop_reason: Optional[Union[int, str]] = None
+    finish_reason: Optional[str] = Field(
+        default='stop')  # per OpenAI spec this is the default
+    stop_reason: Optional[Union[int,
+                                str]] = None  # ??? Not part of the OpenAI spec
```

Comment:
It seems like it's because the engine uses it for something. I asked about it a while ago and was told to create an RFC. I didn't get a lot of feedback, so in this PR I have opted to not change existing behavior in a potentially breaking way even though (imho) it should be removed from the OpenAI entrypoint since it's not part of the spec https://github.com/vllm-project/vllm/issues/6202

Reply:
`stop_reason` will be the token id if there are multiple stop tokens. (You can see this in Llama 3.1, which has multiple stop tokens).